File: fp.cs
using System;

class FunctionPointerDemo
{
    static int Increment(int n)
    {
        return n + 1;
    }
    
    static void MyObjective(int[] arr, Func<int, int> F)
    {
        for (int i = 0; i < arr.Length; ++i)
            arr[i] = F(arr[i]);
    }
    
    static void Main(string[] args)
    {
        int[] arr = {0, 1, 2, 3, 4};
        MyObjective(arr, Increment);
        
        for (int i = 0; i < arr.Length; ++i)
            Console.Write(arr[i] + " ");
    }
}

File: fp.lua
function increment(n)
	return n + 1
end

function my_objective(arr, f)
	for i = 1, 5 do
		arr[i] = f(arr[i])
	end
end

function main()
	local arr = {0, 1, 2 ,3 ,4}
	my_objective(arr, increment)
	for i = 1, 5 do
		io.write(arr[i] .. " ")
	end
end

main()
